### ISSUE TITLE
plugin: Guestful discover plugin

### ISSUE THREAD

**Issue by LecrisUT** [teemtee/tmt#2813]
Does that mean that this plugin is run on the host? For the CMake plugin, I need it to have access to the prepare artifacts since the `ctest` are only known after the configure step (sometimes after build step also). The `ExecutePlugin` seems to be guestful, so I assume this is clashing?

**Comment by happz** [comment]
> The current discover plugin is guestless, does that mean that this plugin is run on the host?

Correct, discovers tests e.g. in the fmf files in the local git repo.

> The ExecutePlugin seems to be guestful

Correct, runs tests on a given guest.

Stinks like a dist-git & rpmbuild scenario again: run discover, run prepare, re-run discover because prepare changed the state of the world somehow. Complicated by the need to access `prepare` artifacts - the rpmbuild runs on the tmt runner, IIUIC? There's a pull from guest to runner, artifacts might be locally available after all. Re-running `discover` is in no solid shape though.

@lukaszachy looks familiar?

**Comment by lukaszachy** [comment]
Painfully familiar. Time to make it generic then and figure out how multihost should work (respect discover phase's where? run it on first guest and copy tests everywhere? ...)

Maybe we can do more usable skeleton as part of #2784 (or at least put some future -ready parts there).

For now it looks that during prepare, after test are discovered, the install require/recommend has to be calculated again. 
There are some common pieces - install tool, run it and let the discover happen.

**Comment by psss** [comment]
@lukaszachy, @LecrisUT, has this been already covered by the latest `dist-git` implementation? Or further steps are needed/desired here?

**Comment by LecrisUT** [comment]
Mostly making the workflow more friendly, e.g. automating this test editing section
https://github.com/teemtee/tmt/blob/29fa59578d469b1ab619440f99e1b9f589764a6a/tmt/steps/discover/fmf.py#L957-L967
